src/sys/socket/multicast.rs

```diff
@@ -37,4 +37,10 @@ impl ipv6_mreq {
             ipv6mr_interface: 0,
         }
     }
+    pub fn new(group: Ipv6Addr, interface: c_uint) -> ipv6_mreq {
```

Comment:
I figured that the sorting out would take some time, and wanted to provide a quick solution. Given that we will probably change the interface in a breaking way anyway in the future, I did not see a problem with that.